.travis.yml

```diff
@@ -2,37 +2,39 @@ language: cpp
 sudo: required
 dist: trusty
 
-compiler:
-  - gcc
+compiler:  # A trick to construct a build matrix
+  - printf "%s" "3.6" >> ~/PYTHON_VERSION ; gcc
+
+os:
+  - linux
+  - osx
+
+env:
+  matrix:
+    - TASK=regular
+    - TASK=mpi
+    - TASK=pylint
+    - TASK=if-else
+    - TASK=sdist
+    - TASK=bdist
+    - TASK=gpu METHOD=source
+    - TASK=gpu METHOD=pip
+    - TASK=regular PYTHON_VERSION=2.7
+    - TASK=regular PYTHON_VERSION=3.4
+    - TASK=regular PYTHON_VERSION=3.5
 
 matrix:
-  include:
-    - os: linux
-      env: TASK=regular
-    - os: linux
-      env: TASK=mpi
-    - os: linux
-      env: TASK=gpu METHOD=source
-    - os: linux
-      env: TASK=sdist
-    - os: linux
-      env: TASK=gpu METHOD=pip
-    - os: linux
-      env: TASK=bdist
-    - os: linux
-      env: TASK=pylint
-    - os: linux
-      env: TASK=if-else
+  exclude:
     - os: osx
-      env: TASK=regular
-    - os: osx
-      env: TASK=mpi
+      env: TASK=gpu METHOD=source
     - os: osx
-      env: TASK=sdist
+      env: TASK=gpu METHOD=pip
     - os: osx
-      env: TASK=bdist
+      env: TASK=pylint
 
 before_install:
+- export DEFAULT_PYTHON=$(< ~/PYTHON_VERSION)
```

Comment:
Is this right?

Reply:
@henry0312 You can go to the line `conda create -q -n test-env python=$PYTHON_VERSION numpy nose scipy scikit-learn pandas matplotlib pytest ` into Travis log and see what version of Python is used.